src/node/blockstorage.h

```diff
@@ -165,6 +165,9 @@ class BlockManager
     //! Returns last CBlockIndex* that is a checkpoint
     const CBlockIndex* GetLastCheckpoint(const CCheckpointData& data) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    /** Best header we've seen so far (used for getheaders queries' starting points). */
+    CBlockIndex* pindexBestHeader = nullptr;
```

Comment:
Is there any reason to move this to blockstorage? It is never used there

Reply:
I think it's used in `BlockManager::AddToBlockIndex` and `BlockManager::LoadBlockIndex`, see previous comment here: https://github.com/bitcoin/bitcoin/pull/22564#discussion_r695990703

Resolve the blurry boundary seems somewhat non-trivial and I don't think would be within the scope of this PR, but we can certainly add a TODO comment so that this doesn't get lost to time? Let me know what you think is best!

Reply:
`LoadBlockIndex` already has a ChainstateManager reference, so this shouldn't cause any issues. See also fab6d6b2d154893ab422dda87f3535d42c3e06f4

How hard would it be to do the same for `AddToBlockIndex`?

Reply:
I'm not sure that reaching up to `ChainstateManager` when loading the block index is the best here... Preferably we can split the "load block index itself" logic from "derive Chainstate variables from loaded block index" logic. Doing so would allow us to keep `pindexBestHeader` in `ChainstateManager` as you suggested (and I think is the right place for it to be in the long run).

Here's a commit that does this: https://github.com/dongcarl/bitcoin/commit/41b61f245f1f115de4625fda40d8898cb6b2d3ba

Branch: https://github.com/bitcoin/bitcoin/compare/master...dongcarl:2022-03-kirby-p2.5

Lmk what you think!

Reply:
Yeah, looks fine long-term. Might want to chat with @jamesob to discuss conflicts.

I just prefer a (temporary) ChainMan reference to be passed down and have the member in ChainMan, instead of (temporarily) putting it in BlockMan. Though, as both are temporary, either is probably fine.

Reply:
No trouble doing this move in terms of planned assumeutxo changes, and certainly having it under blockman is better than prior to this changeset.

I agree some of the "deep-dot" accesses are clunky (probably the necessary result of script-diffs) and there are a few instances where `pindexBestHeader` is accessed like `chainstate.chainman.blockman.pindexbestheader` when it could be `chainstate.blockman.pindexbestheader`, again probably just due to scripted-diffs. If we wanted to ease some of the other uses, we could just have a `ChainstateManager::bestIndex` method that returned from its blockmanager.

Reply:
I think what we're thinking is doing the changes in https://github.com/dongcarl/bitcoin/commits/2022-03-kirby-p2.5, and then making `pindexBestHeader` a `ChainstateManager` member instead of a `BlockManager` one (this current PR makes it a `BlockManager` member).

Reply:
@dongcarl sounds good to me!